tests/FSharp.Compiler.ComponentTests/ErrorMessages/TypeMismatchTests.fs

```diff
@@ -7,6 +7,103 @@ open FSharp.Test.Compiler
 
 module ``Type Mismatch`` =
 
+    module ``Different tuple lengths`` =
+
+        [<Fact>]
+        let ``Known type on the left``() =
+            FSharp """
+let x a b c : int * int = a, b, c
+            """
+            |> typecheck
+            |> shouldFail
+            |> withDiagnostics [
+                (Error 1, Line 2, Col 27, Line 2, Col 34,
+                 "Type mismatch. Expecting a tuple of length 2 of type\n    int * int    \nbut given a tuple of length 3 of type\n    'a * 'b * 'c    \n")
```

Comment:
I've just wondered: would it be feasible to try unifying the types of the matching amount of items and then add fresh type vars for the rest items, so the inferred type here could be `int * int * 'a` instead of `'a * 'b * 'c`? Basically, it means unifying of the items types separately.

Reply:
Wouldn't that run back into the initial issue? Where 
```fsharp
let x : int * int = "a", "b", "c"
```
Would say _Expecting `int * int` but given `int * int * string`_ ?

Reply:
Yes, sounds like it could. I wonder if there's a way to do that only when there's not enough info about the types, but it would be hard to distinguish this situation. The current approach makes it all much better anyway! 🙂